apps/aehyperchains/src/aehc_connector.erl

```diff
@@ -2,96 +2,138 @@
 %%%-------------------------------------------------------------------
 -module(aehc_connector).
 
--export([connector/0]).
+-export([send_tx/4, get_block_by_hash/2, get_top_block/1]).
 
--export([send_tx/1, get_block_by_hash/1, get_top_block/0]).
--export([tx/2, block/3]).
--export([publish_block/1, subscribe_block/0]).
+-export([commitment/2, parent_block/4]).
+-export([publish_block/2]).
+-export([module/1]).
+-export([args/1]).
+
+-export([accept/1]).
 
 -type connector() :: atom().
 
--callback send_tx(binary()) -> binary().
+-type commitment() :: aehc_commitment:commitment().
+-type parent_block() :: aehc_parent_block:parent_block().
 
--callback get_top_block() -> block().
+-callback send_tx(binary(), binary(), binary()) -> ok.
+-callback get_top_block() -> parent_block().
+-callback get_block_by_hash(binary()) -> parent_block().
 
--callback get_block_by_hash(binary()) -> block().
+-export_type([connector/0]).
 
 %%%===================================================================
 %%%  Parent chain simplified proto
 %%%===================================================================
-
--record(tx, { sender_id :: binary(), payload :: binary() }).
-
--type tx() :: #tx{}.
-
--record(block, { hash :: binary(), prev_hash :: binary(), txs :: [tx()] }).
-
--type block() :: #block{}.
-
--export_type([tx/0, block/0]).
-
--spec tx(Sender::binary(), Payload::binary()) -> tx().
-tx(SenderId, Payload) when
-      is_binary(SenderId), is_binary(Payload) ->
-    #tx{ sender_id = SenderId, payload = Payload }.
-
--spec block(Hash::binary(), PrevHash::binary(), Txs::[tx()]) -> block().
-block(Hash, PrevHash, Txs) when
-      is_binary(Hash), is_binary(PrevHash), is_list(Txs) ->
-    #block{ hash = Hash, prev_hash = PrevHash, txs = Txs }.
+%% This API should be used by connector's developer as a wrapper around internal abstract commitments data type;
+-spec commitment(Delegate::binary(), KeyblockHash::binary()) ->
+    commitment().
+commitment(Delegate, KeyblockHash) when
+    is_binary(Delegate), is_binary(KeyblockHash) ->
+    aehc_commitment:new(aehc_commitment_header:new(Delegate, KeyblockHash)).
+
+%% TODO: Opened for discussion with gorbak25 and radrow;
+%%-spec commitment(Delegate::binary(), KeyblockHash::binary(), PoGFHash::binary()) ->
+%%    commitment().
+%%commitment(Delegate, KeyblockHash, PoGFHash) when
+%%    is_binary(Delegate), is_binary(KeyblockHash), is_binary(PoGFHash) ->
+%%    aehc_commitment:new(aehc_commitment_header:new(Delegate, KeyblockHash, PoGFHash)).
+
+-spec parent_block(Height::non_neg_integer(), Hash::binary(), PrevHash::binary(), Commitments::[commitment()]) ->
+    parent_block().
+parent_block(Height, Hash, PrevHash, Commitments) when
```

Comment:
Ok, this helper is still broken and generates an invalid parent block ADT.
Please preserve the following invariant:
```
is_list(Block.Header.commitment_hashes),
is_list(Block.commitments),
length(Block.Header.commitment_hashes) == length(Block.commitments),
[Hash = aehc_commitment:hash(Commitment) || {Hash, Commitment} <- lists:zip(Block.Header.commitment_hashes, Block.commitments)]
```
This is essentially the same invariant we preserve when handling microblocks. 
Take a look at:
https://github.com/aeternity/aeternity/blob/8b33ad36d6f81d90f57178b735819c01bfe652c8/apps/aehyperchains/src/aehc_parent_db.erl#L96

Reply:
How much beer have you drunk yesterday that posted Elixir syntax here? Joke :) Missed one argument in constructor, thanks for the notice.